webhooks-extension/test/util.sh

```diff
@@ -126,69 +126,103 @@ function install_tekton() {
+    dashboard_dir=$GOPATH/src/github.com/tektoncd/dashboard
+    if [ -d $dashboard_dir ]; then
+      # Notify and depending on [version] we will ko apply or kubectl apply gcr yaml
+      echo -e "\n\nDashboard source detected at $dashboard_dir"
+    else
+      echo -e "\n\nDashboard source not detected in $dashboard_dir"
+      if [[ $version != "3" ]]; then
+        read -p "Do you wish to git clone and ko apply the master version? " -n 1 -r cloneAndApply
+      fi
+      if [[ $cloneAndApply =~ ^[Yy]$ ]] || [[ $version == "3" ]]; then
+        pushd $GOPATH/src/github.com/tektoncd/
+        git clone https://github.com/tektoncd/dashboard.git
+        popd
+      else
+        # User does not want to clone repo therefore if user chooses to use the
+        # nightly build then the only way to get it is by curling it
+        echo -e "\nCurling and applying nightly build...."
+        curl -L https://github.com/tektoncd/dashboard/releases/download/v0/gcr-tekton-dashboard.yaml | kubectl apply -f -
+      fi
+    fi
+
+    # At this point has the dashboard cloned and the only thing left to do is
+    # ko apply or kubectl apply gcr yaml
+    if [[ $version == "3" ]]; then
+      pushd $GOPATH/src/github.com/tektoncd/dashboard
+      docker login
+      npm ci
+      npm run build_ko
+      dep ensure -v
+      timeout 60 "ko apply -f config -n $namespace"
+      popd
+
+    elif [[ $cloneAndApply =~ ^[Yy]$ ]] || [[ -d $dashboard_dir ]]; then
+      kubectl apply -f config/release/gcr-tekton-dashboard.yaml
+    fi
   fi
-  if [[ $cloneAndApply =~ ^[Yy]$ ]]; then
-    pushd $GOPATH/src/github.com/tektoncd/
-    git clone https://github.com/tektoncd/dashboard.git
-    popd
-    apply="y"
+}
+
+# Install webhooks extension (https://github.com/tektoncd/experimental/tree/master/webhooks-extension)
+#
+# Installation process is either:
+# kubectl applying gcr current releaseyaml
+# kubectl applying gcr nightly build yaml
+# ko apply development version
+function install_webhooks_extension() {
+  if [ -z "$1" ] || [ -z "$2" ]; then
+      echo "Usage ERROR for function: install_webhooks_extension [namespace] [version] <sleepTime>"
+      [ -z "$1" ] && echo "Missing [namespace]"
+      [ -z "$2" ] && echo "Missing [version]"
+      exit 1
   fi
-  if [[ $apply =~ ^[Yy]$ ]]; then
-    pushd $GOPATH/src/github.com/tektoncd/dashboard
+
+  version=$2
+
+  # Current Release
+  if [[ $version == "1" ]]; then
+    kubectl apply --filename https://github.com/tektoncd/dashboard/releases/latest/download/webhooks-extension_release.yaml
+  # Nightly Build
+  elif [[ $version == "2" ]]; then
+    pushd $GOPATH/src/github.com/tektoncd/experimental/webhooks-extension
+    kubectl apply -f config/latest/gcr-tekton-webhooks-extension.yaml
+    popd
+  # Development Version
+  else
+    pushd $GOPATH/src/github.com/tektoncd/experimental/webhooks-extension
+    namespace=$1
     docker login
-    npm install
+    npm ci
+    npm rebuild node-sass
```

Comment:
Is this actually needed? Haven't done it in months

Reply:
great point. will remove and see if everything works properly!

Reply:
`npm rebuild` should only be needed (occasionally) when switching node versions